catalog/spatial/geocoding/spatial-geocoding-plugin/src/main/java/org/codice/ddf/spatial/geocoding/plugin/GeoCoderPlugin.java

```diff
@@ -118,11 +110,33 @@ public int getRadius() {
         return radiusInKm;
     }
 
-    private boolean countryCodeNeeded(Metacard metacard) {
-        return metacard.getAttribute(Location.COUNTRY_CODE) == null;
+    /**
+     * Sets the country code attribute of the {@param metacard} when the {@param metacard} has a
+     * location and there is not already a country code attribute. Does not update the country code
+     * if the country code attribute is already set.
+     *
+     * @param metacard
+     */
+    private void setCountryCode(Metacard metacard) {
+        Optional<String> wktLocation = Optional.ofNullable(metacard.getLocation());
+
+        if (geoCoder != null && wktLocation.isPresent() && !hasCountryCode(metacard)) {
+            Optional<String> alpha3CountryCode = geoCoder.getCountryCode(wktLocation.get(),
+                    radiusInKm);
+
+            geoCoder.getCountryCode(wktLocation.get(), radiusInKm)
+                    .ifPresent(countryCode -> {
+                        metacard.setAttribute(new AttributeImpl(Location.COUNTRY_CODE,
+                                alpha3CountryCode.get()));
+
+                        LOGGER.trace("Setting metacard country code to {} for metacard with id {}",
```

Comment:
Could you move the trace statement before the operation it is describing? That way if it fails, the log will show what is trying to be done when it fails. Otherwise, it will only print out if it was successful.